### PR TITLE
BUG: Fix incorrect output extent of volume reconstructor

### PR DESCRIPTION
**Pull request by jrojasUNC**
Change the output spacing slightly so the output extent contains all frames without exceeding the voxel lattice. The output extent can extend beyond existing data when using ceil, or not include some of the data when using floor, which produces holes in the image.

The frame spacing changed only by about 0.1% during my testing, but I expect it depends on the precision of the encoders.

This is a follow up to https://github.com/IGSIO/IGSIO/pull/38. Please let me know what you think @Sunderlandkyl , @dzenanz.

### CODE REVIEW

**Review by dzenanz — COMMENTED** [review]
Would it be problematic to change output spacing if it was specified by the user? Otherwise I am fine with the change. I did not look too hard into the computation math,

**Comment by jrojasUNC** [comment]
I don't believe so because the change is very small. It's really just a nudge to account for any noise in encoder values. To put the 0.1% I mentioned into real numbers, the frame spacing was changed from 1.60148 - 1.60275 in one of my volumes.

**Comment by dzenanz** [comment]
What I had in mind was changing `2.0` into `1.999`, which might bother someone.

**Comment by lassoan** [comment]
If the user set a specific output spacing then that value must be used. It does not matter how small the change is, we cannot change it. If some new automatic spacing setting option is added (where the user just specifies some hints for the spacing and not an exact value) then the software would be allowed to set/adjust the spacing value.

There should be no problem with tuning automatic extent computation algorithm. It could make sense to round the values or use `ceil` with some tolerance. For example, we could change to:

    tolerance = 0.1; // extent is only expanded if the voxel is filled by at least 10%
    outputExtent[d * 2 + 1] = int(std::ceil((extent_Ref[d * 2 + 1] - outputOrigin_Ref[d]) / outputSpacing[d] - tolerance));

File: VolumeReconstruction/vtkIGSIOVolumeReconstructor.h [hunk 1]
```diff
@@ -125,6 +125,10 @@ class VTKVOLUMERECONSTRUCTION_EXPORT vtkIGSIOVolumeReconstructor : public vtkIma
   vtkGetMacro(EnableFanAnglesAutoDetect, bool);
   vtkSetMacro(EnableFanAnglesAutoDetect, bool);
 
+  /*! Get/set RecomputeOutputSpacing. if true, recompute output spacing so reconstruction grid matches data exactly. */
+  vtkGetMacro(RecomputeOutputSpacing, bool);
+  vtkSetMacro(RecomputeOutputSpacing,bool)
```
Comment (jamesobutler) [inline VolumeReconstruction/vtkIGSIOVolumeReconstructor.h:130]:
```suggestion
  vtkSetMacro(RecomputeOutputSpacing, bool);
```

Comment (jrojasUNC) [reply VolumeReconstruction/vtkIGSIOVolumeReconstructor.h]: Thank you. Fixed.

**Comment by jrojasUNC** [comment]
@lassoan can you take a look at this when you have a moment?

**Comment by jrojasUNC** [comment]
@lassoan @Sunderlandkyl Do you want me to make any more changes here?

**Review by jamesobutler — APPROVED** [review]
@Sunderlandkyl Can you help with the integration of this PR? Thanks!

**Pull request merged**